### PR TITLE
Interest Rate Boundary Conditions Testing

### LINKED ISSUE #180: Interest Rate Boundary Conditions Testing
## Context
Tests cover normal utilization ranges, need extreme scenarios.

**Missing Test Coverage:**

- **Extreme Utilization**
  - Utilization = 99.9% (nearly all liquidity borrowed)
  - Interest rate at maximum cap (400%)
  - KinkCurve steep slope behavior
  - Borrowing cost discourages further borrowing

- **Zero Balance Edge Cases**
  - totalCreditBalance = 0 (no liquidity)
  - Attempting to borrow should fail
  - Interest rate calculation with zero denominator
  - New deposit enables borrowing

- **Empty Pool**
  - totalDebitBalance = 0 (no borrows)
  - Credit interest rate = 0
  - First borrow triggers rate update
  - Utilization jumps from 0% to X%

- **Kink Point Transitions**
  - Utilization exactly at kink (e.g., 80.0%)
  - Rate calculation uses slope1 or slope2?
  - Smooth transition validation
  - No discontinuities or jumps

- **Maximum Rate Enforcement**
  - Interest curve returns 450% (exceeds cap)
  - Protocol enforces 400% maximum
  - Validation in postcondition
  - No overflow or panic

- **Long Time Period Accrual**
  - Interest accrues for 1 year (31,557,600 seconds)
  - Interest accrues for 10 years
  - Compound interest doesn't overflow
  - UFix128 precision maintained

- **Time Jump Scenarios**
  - Blockchain halts for 1 day
  - First transaction after restart
  - Interest accrual for large time delta
  - No overflow in `dt` calculation

**Recommended Tests:**
```
Test: Utilization 99.9%, validate interest rate
Test: totalCreditBalance = 0, borrow attempt fails
Test: totalDebitBalance = 0, credit rate = 0
Test: Utilization exactly 80.0% (kink point)
Test: Interest curve returns 450%, enforced to 400%
Test: Accrue interest for 10 years, no overflow
Test: Time jump 1 day, interest accrual correct
```

### PR DESCRIPTION
**Pull request by UlyanaAndrukhiv**
Closes: #180

## Context
- Added `fork_interest_rate_test.cdc`, a new `mainnet-fork` test suite covering `KinkCurve` behavior across different scenarios.
- Added explicit `swapperInTypeIdentifier / swapperOutTypeIdentifier` parameters to `setInsuranceSwapper` helper.
- Added `receiverVaultStoragePath` parameter to `withdrawFromPosition` helper, making it return the transaction result instead of asserting internally.
- Added support for any token type to `withdraw_from_position transaction`, dropping the hardcoded `FlowToken` path.
- Updated `MAINNET_FLOW_HOLDER_ADDRESS` constant

### CODE REVIEW

**Review by holyfuchs — APPROVED** [review]

**Review by mts1715 — COMMENTED** [review]
few smal fixes should be done

**Review by mts1715 — APPROVED** [review]

File: cadence/tests/test_helpers.cdc [hunk 1]
```diff
@@ -1065,15 +1066,23 @@ fun withdrawReserve(
...
 
 access(all) fun equalWithinVariance(_ expected: AnyStruct, _ actual: AnyStruct, _ variance: AnyStruct): Bool {
-    let expectedType = expected.getType()
-    let actualType = actual.getType()
-    let varianceType = variance.getType()
-    if expectedType == Type<UFix64>() && actualType == Type<UFix64>() && varianceType == Type<UFix64>() {
-        return ufixEqualWithinVariance(expected as! UFix64, actual as! UFix64, variance as! UFix64)
-    } else if expectedType == Type<UFix128>() && actualType == Type<UFix128>() && varianceType == Type<UFix128>(){
-        return ufix128EqualWithinVariance(expected as! UFix128, actual as! UFix128, variance as! UFix128)
+    // Try UFix64 first
```
Comment (zhangchiqing) [inline cadence/tests/test_helpers.cdc:1069]: Fixed equalWithinVariance in test_helpers.cdc - Replaced .getType() calls with optional binding (as?) since
  primitives like UFix64 don't have a .getType() method in Cadence.
